Jetpack Disconnect: Add Tracks event to beginning of Flow

ockham opened the pull request:
Counterpart to #19164. Track whenever a user first starts the disconnect flow (which right now only consists of the confirmation dialog) so we have a funnel to see what percentage actually follows thru and compare that to the [new flow](https://github.com/Automattic/wp-calypso/projects/49) we're currently implementing.

To test:
* In your browser console, enter `localStorage.setItem( 'debug', 'calypso:analytics*' )`, and reload.
* Verify that the `calypso_jetpack_disconnect_start` Tracks event is fired in the following cases:

1. `/settings/manage-connection/:site` -- Click the 'Disconnect from WordPress.com' button at the bottom of the page

![image](https://user-images.githubusercontent.com/96308/31471251-3b2258a6-aee8-11e7-87e6-ece683dccf69.png)

2. In siteslist in the sidebar, if you have a “broken” jetpack site, it will let you disconnect from the site there. (For example, rename `jetpack` plugin folder on the site.)

<img width="273" alt="screen shot 2017-06-23 at 12 41 17" src="https://user-images.githubusercontent.com/7767559/27480757-4ad41862-5811-11e7-8036-64202338667c.png">


Question: Did I miss any entry points for the current Disconnect flow?

sirreal commented:
This may be small, but I notice slightly different behavior.

`/settings/manage-connection/:site` When I click here, I see the tracks immediately, as the modal appears.

![image](https://user-images.githubusercontent.com/96308/31471251-3b2258a6-aee8-11e7-87e6-ece683dccf69.png)

2. In siteslist in the sidebar- when I click here, I see the tracks when I _close_ the modal… even when I pick "stay connected" odd.

<img width="273" alt="screen shot 2017-06-23 at 12 41 17" src="https://user-images.githubusercontent.com/7767559/27480757-4ad41862-5811-11e7-8036-64202338667c.png">

ockham commented:
> In siteslist in the sidebar- when I click here, I see the tracks when I close the modal… even when I pick "stay connected" odd.

That's because I had added the tracks event to the `hideDialog` method :man_facepalming: 
Fixed by 4e036d6. Sorry for the sloppiness, filed this PR quickly before hopping on a call. Weak excuse, I know 😄
